cli/src/semgrep/config_resolver.py

```diff
@@ -253,6 +261,91 @@ def _load_config_from_local_path(self) -> List[ConfigFile]:
     def is_registry_url(self) -> bool:
         return self._origin == ConfigType.REGISTRY
 
+    def _project_metadata_for_standalone_scan(self) -> out.ProjectMetadata:
+        repo_name = os.environ.get("SEMGREP_REPO_NAME")
+        if repo_name is None:
```

Comment:
@jbergler How necessary is the `repo_name`? This check no longer makes it possible to run `semgrep scan --config supply-chain` (supply chain was working before, and same applies to secrets). 

If we're requiring it be defined, can we pass "unknown"? or figure out a different way to grab repo name?

Reply:
Looks like `SEMGREP_REPO_NAME` is [semgrep ci specific](https://semgrep.dev/docs/semgrep-ci/ci-environment-variables/#semgrep_repo_name). Does it make sense to skip this check (or is ConfigLoader used for `ci` as well in which case we skip the check for `scan` only)?

Reply:
might take a peek at `GitMeta` as another flavor of pulling `repo_name` using `git rev-parse` – 
https://github.com/semgrep/semgrep/blob/b47ebc9c0a914b669addc069e0160b3f5193a80c/cli/src/semgrep/meta.py#L103

Reply:
`semgrep scan` can't have git dependencies as it stands, so I'd rather just limit requiring the env var if code or secrets are being requested, and we'll figure out auto detection / adding it for supply chain later.

eg:

```suggestion
        if repo_name is None and require_repo_name:
```

Reply:
Done in [fc41789](https://github.com/semgrep/semgrep/pull/9205/commits/fc41789b107c847fdf58199b621316a77509f6bc). 

`--config <secrets|code>` requires SEMGREP_REPO_NAME.